### ISSUE TITLE
Borg running full backups on only one specific mount daily 

### ISSUE THREAD

**Issue by mysterious-maniac** [borgbackup/borg#8351]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Question

#### Your borg version (borg -V).

borg 1.1.17

#### Operating system (distribution) and version.

RHEL 8.6

#### How much data is handled by borg?

1.5T

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

`ssh -A -q -o StrictHostKeyChecking=no ${BK_USER}@${BK_HOST} "BORG_PASSPHRASE=\$(cat) borg --rsh 'ssh -q -o StrictHostKeyChecking=no' create ${BORG_CREATE_OPTIONS} ssh://${BORG_USER}@${BORG_HOST}${BORG_REPO}::${BK_TYPE}-{now} ${BK_PATH}";`

## Describe the problem you're observing.

On one of our environments, borg backup runs a **_full_** backup of a specified mount daily instead of what should be only any new files/changes to files stored on that mount.

The above commands is from a script, which runs daily via crond. 

The mount (/mountX) being backed up in question is ~580G, meaning that the backup process for this one environment takes 12+ hours daily. There are various other mountpoints that are backed up daily along with it, but those seem to be working fine with only new changes being backed up.

I read through https://github.com/borgbackup/borg/issues/6089 and the cache for this environment already exists from the previous day's backup, so I don't understand why /mountX has every single file backed up again daily.

#### Include any warning/errors/backtraces from the system logs

No errors, but this is an example of the time taken to perform the daily backup
```
------------------------------------------------------------------------------
Archive name: daily-2024-09-02T08:41:03
Archive fingerprint: c3a6b88dfb20d645ba8cfe39dc7dc08fa261063bf52efa45c4abb7873c2f4df1
Time (start): Mon, 2024-09-02 08:41:05
Time (end):   Tue, 2024-09-03 01:20:20
Duration: 16 hours 39 minutes 14.88 seconds
Number of files: 23282
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:              630.66 GB            496.37 GB             18.56 MB
All archives:               43.93 TB             34.26 TB            327.98 GB

                       Unique chunks         Total chunks
Chunk index:                  174854             18729724
------------------------------------------------------------------------------
```

**Comment by ThomasWaldmann** [comment]
Check the FAQ about how to debug files cache issues.

Usually it is because:
- full path changed (mountpoint changed)
- file ctime changed (e.g. due to file metadata updates)
- inode numbers changed (some fs do not offer stable inode numbers)
- doing more than files cache TTL **different** backup runs

**Comment by ThomasWaldmann** [comment]
Also, check if you can use a more recent borg version like 1.2.x or 1.4.x. Read the changelog before upgrading.

**Comment by mysterious-maniac** [comment]
Thanks for the quick response.

The culprit was indeed 'file ctime changed'.

I found a rogue cronjob which chown'ed the problematic mountpoint nightly, which didn't change the timestamp when running an 'ls -l' but when checking the ctime with 'ls -lc', it led me to the cronjob.

I've since disabled the cronjob and this will likely solve the issue. 

Additional related issues I found which were helpful for future readers:
https://github.com/borgbackup/borg/issues/7767
https://github.com/borgbackup/borg/issues/7931

Thanks again!
